### PR TITLE
Added flags parameter to PGTBL CAPTBL_OP_CPY

### PR DESCRIPTION
**Pull request by evanstella**
### Summary of this Pull Request (PR)

Added flags parameter to PGTBL CAPTBL_OP_CPY to edit PTE flags to allow delegation of page access rights. I had to change quite a bit of `u32_t` to `word_t` in the CHAL code since 64bit paging on x86_64 uses some high qword bits in the PTE as flags. I think this is also a better design since the x86 CHAL supports both 32bit and 64bit execution modes, but I might be wrong so I am definitely looking for feedback on those edits.

### Intent for your PR

Choose one (Mandatory):

- [X] This PR is for a code-review and is intended to get feedback, but not to be pulled yet.
- [ ] This PR is mature, and ready to be integrated into the repo.

### Reviewers (Mandatory):
(Specify @<github.com username(s)> of the reviewers. Ex: @user1, @user2)
@gparmer @WenyuanShao @betahxy

### Code Quality

As part of this pull request, I've considered the following:

[Style](https://github.com/gparmer/composite/raw/ppos/doc/style_guide/composite_coding_style.pdf):

- [X] Comments adhere to the Style Guide (SG)
- [X] Spacing adhere's to the SG
- [X] Naming adhere's to the SG
- [X] All other aspects of the SG are adhered to, or exceptions are justified in this pull request
- [ ] I have run the auto formatter on my code before submitting this PR (see doc/auto_formatter.md for instructions)

[Code Craftsmanship](http://www2.seas.gwu.edu/~gparmer/posts/2016-03-07-code-craftsmanship.html):

- [X] I've made an attempt to remove all redundant code
- [X] I've considered ways in which my changes might impact existing code, and cleaned it up
- [X] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [X] I've commented appropriately where code is tricky
- [X] I agree that there is no "throw-away" code, and that code in this PR is of high quality

### Testing

I've tested the code using the following test programs (provide list here):

- [ ] micro_booter
- [X] unit_pingpong
- [X] unit_schedtests
- [x] ...sched_ping_pong, sched_crt_tests

### CODE REVIEW

File: src/platform/i386/chal/shared/cos_config.h [hunk 1]
```diff
@@ -164,6 +164,15 @@
                             /* 30/1G   31/2G */ \
                                -1,     -1 \
 
+/* PTE user pagetable modifiable flags */
+#define COS_PAGE_READABLE (0)
+#define COS_PAGE_WRITABLE (1ul << 1)
+#define COS_PAGE_PKEY0    (1ul << 59)
```
Comment (gparmer) [inline src/platform/i386/chal/shared/cos_config.h:170]: A note for future github searches: we likely don't want to use these values in most code. Doing so will make the corresponding code non-portable.

**Review by gparmer — APPROVED** [review]
I am beyond happy with this. I can't thank you enough for pushing this through. I assume it is tested, and works for component execution. TBH, I'm a little surprised it didn't catch some off-by N errors that would have written to now RO memory. Great job with many of the kernel updates. I can't believe that many addresses were u32s! Yikes.

I'd like @betahxy to review as he is most familiar with many of the kernel changes, but I'm happy with it.

File: src/components/lib/kernel/cos_kernel_api.c [hunk 2]
```diff
@@ -1062,7 +1062,7 @@ cos_rcv(arcvcap_t rcv, rcv_flags_t flags, int *rcvd)
 }
 
 vaddr_t
-cos_mem_aliasn(struct cos_compinfo *dstci, struct cos_compinfo *srcci, vaddr_t src, size_t sz)
+cos_mem_aliasn(struct cos_compinfo *dstci, struct cos_compinfo *srcci, vaddr_t src, size_t sz, unsigned long newflags)
```
Comment (betahxy) [inline src/components/lib/kernel/cos_kernel_api.c:1065]: @gparmer do we need a new name for this `newflags`? I feel that this name is not that easy to understand its meaning

Comment (gparmer) [reply src/components/lib/kernel/cos_kernel_api.c:1065]: Good comment.  `perm_flags`? I think that most would associated `perm` with "permissions"?

Comment (evanstella) [reply src/components/lib/kernel/cos_kernel_api.c]: Agreed, made the change.

**Comment by gparmer** [comment]
Thanks!

**Pull request merged**